Allow PHP 8

ddeboer opened the pull request:
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| Documentation   | 
| License         | MIT


#### What's in this PR?

Support PHP 8.0.

ddeboer committed:
Allow PHP 8

ddeboer committed:
Use newer PHPSpec to fix namespace issue

ddeboer committed:
Remove short_array_syntax

"Arrays MUST use the short syntax."
https://docs.styleci.io/fixers#short_array_syntax

ddeboer committed:
Clean up testing matrix

dbu commented on composer.json:
```diff
@@ -11,12 +11,12 @@
     ],
     "require": {
-        "php": "^7.1",
+        "php": ">=7.1",
```
i'd prefer to not do this, it might create hassles when php 9 comes out. 

lets do ^7.1 || ^8.0

or 7.2 or whatever. i think the rest of the php-http currently supports 7.2

can we allow some older phpspec version that works on 7.2? there seem to have been no changes needed to the spec files, so it should be cheap to allow it.

ddeboer replied:
Sure, there’s quite some discussion about this still: https://twitter.com/nicolasgrekas/status/1263023258938548225.

I’m okay with narrowing the version constraint, but decided for `^7.3`, which is the earliest supported PHP version.

dbu reviewed (changes requested):
thanks for looking into this

ddeboer committed:
Narrow PHP version constraint

ddeboer committed:
Remove custom dependency

ddeboer committed:
Run coverage on 7.4

ddeboer committed:
Remove code coverage monitoring

"CodeCoverage should not be used with spec testing in order to see how good your tests are."

https://github.com/henrikbjorn/PhpSpecCodeCoverageExtension

ddeboer commented on .travis.yml:
```diff
@@ -17,14 +17,13 @@ matrix:
-        - php: 7.1
-        - php: 7.2
         - php: 7.3
-          env: COVERAGE=true TEST_COMMAND="composer test-ci" DEPENDENCIES="henrikbjorn/phpspec-code-coverage:^1.0"
```
This may need some discussion, but this dependency is [deprecated](https://github.com/henrikbjorn/PhpSpecCodeCoverageExtension) so I decided to remove code coverage.

dbu replied:
fine for me

dbu reviewed (approved):
thanks.

i strongly disagree with nikolas on that one. its trivial to open a new major php version when it comes along, if there are no BC breaks affecting the library. if there are BC breaks, its a pain for users to figure out if they will have a problem when they use dependencies that don't properly declare what they are tested with. i keep closing issues on our guzzle6-adapter from people asking to support php 8, when guzzle 6 itself has failing tests on php 8 (but uses the >= 5.5 constraint [note how guzzle 7 properly declares php versions it is compatible with])

dbu commented:
tagged a release https://github.com/php-http/stopwatch-plugin/releases/tag/1.4.0
